src/cls/rgw/cls_rgw_types.h

```diff
@@ -1226,7 +1226,7 @@ struct cls_rgw_lc_obj_head
     uint64_t t = start_date;
     encode(t, bl);
     encode(marker, bl);
-    encode(shard_rollover_date, bl);
+    encode((uint64_t)shard_rollover_date, bl);
```

Comment:
in general, we can't safely change the encoded size of a field without bumping the encoding version and handling decode of both the new and existing versions

for `time_t` specifically, this is probably harmless? if any platform was encoding a 32-bit `time_t` here, it would fail to decode that as a `uint64_t`. so this change shouldn't break anything that was working before

what do you think @mattbenjamin?